pkg/subctl/cmd/join.go

```diff
@@ -441,6 +450,19 @@ func populateSubmarinerSpec(subctlData *datafile.SubctlData) submariner.Submarin
 	if globalCIDR != "" {
 		submarinerSpec.GlobalCIDR = globalCIDR
 	}
-
 	return submarinerSpec
 }
+
+func createSAPerCluster(clientset *kubernetes.Clientset, brokernamespace string) *v1.Secret {
+	newsa, err := broker.CreateNewBrokerSA(clientset, clusterID+"-submariner-k8s-broker-client")
+	exitOnError("Service account not created", err)
+	newrole, err := broker.CreateNewBrokerRole(clientset, newsa.Name)
```

Comment:
We don't need to keep a new role for every new cluster. You can have a single role, and then using role bindings to associate the SA to the Role.

Also in this case you'd need to use the explained above "NewClusterBrokerRole", to ensure at least that common role is created.

Reply:
I tried to not create a new role for every cluster but instead have all clusters bind to the same role. This somehow didn't work. I think we need to group SAs and then bind the role to that group. Searching for syntax to do this.

Reply:
I mean something like this: 
```
[majopela@bluehat submariner]$ kind create cluster
Creating cluster "kind" ...
 ✓ Ensuring node image (kindest/node:v1.15.3) 🖼 
 ✓ Preparing nodes 📦 
 ✓ Creating kubeadm config 📜 
 ✓ Starting control-plane 🕹️ 
 ✓ Installing CNI 🔌 
 ✓ Installing StorageClass 💾 
Cluster creation complete. You can now use the cluster with:
export KUBECONFIG="$(kind get kubeconfig-path --name="kind")"
kubectl cluster-info
[majopela@bluehat submariner]$ export KUBECONFIG="$(kind get kubeconfig-path --name="kind")"
[majopela@bluehat submariner]$ kubectl create role pod-reader --verb=get --verb=list --verb=watch --resource=pods
role.rbac.authorization.k8s.io/pod-reader created
[majopela@bluehat submariner]$ kubectl create sa sa-1
serviceaccount/sa-1 created
[majopela@bluehat submariner]$ kubectl create sa sa-2
serviceaccount/sa-2 created
[majopela@bluehat submariner]$ kubectl create rolebinding sa-1-pod-reader --role=pod-reader --serviceaccount=default:sa-1
rolebinding.rbac.authorization.k8s.io/sa-1-pod-reader created
[majopela@bluehat submariner]$ kubectl create rolebinding sa-2-pod-reader --role=pod-reader --serviceaccount=default:sa-2
rolebinding.rbac.authorization.k8s.io/sa-2-pod-reader created
```